src/Microsoft.AspNetCore.Mvc.Razor.Extensions/RazorPageDocumentClassifierPass.cs

```diff
@@ -58,6 +73,32 @@ private void EnsureValidPageDirective(RazorCodeDocument codeDocument)
                 pageDirective.DirectiveNode.Diagnostics.Add(
                     RazorExtensionsDiagnosticFactory.CreatePageDirective_CannotBeImported(pageDirective.DirectiveNode.Source.Value));
             }
+            else
+            {
```

Comment:
I'd add a bigger disclaimer at the top of this about why we're re-parsing the document (consistency with MVC runtime + can't see the parse tree)

Reply:
Made the comment better.